editor/editor_properties_array_dict.cpp

```diff
@@ -517,6 +539,52 @@ void EditorPropertyArray::drop_data_fw(const Point2 &p_point, const Variant &p_d
 		emit_changed(get_edited_property(), array, "", false);
 		update_property();
 	}
+
+	if (drop_type == "nodes") {
+		Array nodes = drag_data["nodes"];
+		Node *base_node = get_base_node();
+
+		for (int i = 0; i < nodes.size(); i++) {
+			Node *node = get_node(nodes[i]);
+			NodePath path = node->get_path();
+
+			if (subtype == Variant::OBJECT) {
+				if (!base_node) {
+					array.call("push_back", get_tree()->get_edited_scene_root()->get_node(path));
+				} else {
+					array.call("push_back", base_node->get_node(path));
+				}
```

Comment:
here you are doing some useless back and forth you already have the node so you don't need to pass through the path doing so will just create ways for it to fail and is longer in term of code and execution for no reasons.
```suggestion
			if (subtype == Variant::OBJECT) {
				array.call("push_back", node);
```